src/Syntax/Term.hs

```diff
@@ -97,6 +100,32 @@ cata var alg = go where
     Var v -> var v
     Alg t -> alg (hmap go t)
 
+cataM
+  :: forall sig m f a
+  .  ( HTraversable sig
+     , Monad m
+     )
+  => (forall x . x -> m (f x))
+  -> (forall x . sig f x -> m (f x))
+  -> (Term sig a -> m (f a))
+cataM var alg = go where
+  go :: forall a . Term sig a -> m (f a)
+  go = \case
+    Var v -> var v
+    Alg t -> alg =<< htraverse go t
```

Comment:
`cata` and `cataM` are interdefinable, however:

1. Defining `cataM` in terms of `cata` requires a `Traversable m` constraint, which won’t be satisfiable for anything using e.g. `ReaderC` or `StateC`.

2. Defining `cata` in terms of `cataM` requires an `HTraversable sig` constraint. This is very likely to be available for syntax (you can derive it, after all), but still a stronger constraint than we otherwise need; as written, `cata` accommodates `fused-effects`’ extant effects, whereas `cataM` would require one to define (orphan) `HTraversable` instances, which isn’t feasible for many effects (any of them containing functionals).

IMO this justifies the separate definitions.